### DISCUSSION TITLE
Supercluster in composite layer = sometimes double highlights

### DISCUSSION THREAD

**Discussion by ImDino** [visgl/deck.gl#6766] [Q&A]
I have a Vue 3 app with a composite layer to use supercluster.

The problem is that when I have two sublayers (one for single points, see below) some icons have the same index number, and both get highlighted (they aren't on top of eachother or anything btw).

`
return [
        new IconLayer(
          this.getSubLayerProps({
            data: this.state.data.filter(i => i.properties.cluster),
            ...
        ),
        new IconLayer(
          this.getSubLayerProps({
            data: this.state.data.filter(i => !i.properties.cluster),
            ...
        )
      ];
`

How do I circumvent this?

Thanks

**Comment by ted341** [comment]
@ImDino - any solution for this?

**Reply by ImDino** [reply]
I'm sorry to say, but I haven't used it since. I haven't been coding much for a long time now tbh. I hope you find a solution.

**Reply by ted341** [reply]
@ImDino - np, thanks for your reply anyway
